packages/typespec-client-generator-core/src/package.ts

```diff
@@ -390,7 +388,7 @@ function getSdkMethodResponse(
       name: createGeneratedName(context, operation, "UnionResponse"),
       isGeneratedName: true,
       clientNamespace: client.clientNamespace,
-      crossLanguageDefinitionId: getCrossLanguageDefinitionId(context, operation),
+      crossLanguageDefinitionId: `${getCrossLanguageDefinitionId(context, operation)}.UnionResponse`,
```

Comment:
So, for these kind of changes, we are making a `crossLanguageDefinitionId` that being cross language in SDK, but does not exist in TypeSpec itself?

Reply:
yes. for these tcgc generated types, we create the definition id. maybe in efv2, we will try to create native typespec type, then we could do the mapping.